common/persistence/serialization/codec.go

```diff
@@ -0,0 +1,69 @@
+package serialization
+
+import (
+	"errors"
+	"reflect"
+
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/server/common/codec"
+	"google.golang.org/protobuf/proto"
+)
+
+func ProtoEncode(m proto.Message) (*commonpb.DataBlob, error) {
+	return encodeBlob(m, enumspb.ENCODING_TYPE_PROTO3)
+}
+
+func encodeBlob(m proto.Message, encoding enumspb.EncodingType) (*commonpb.DataBlob, error) {
+	if m == nil || (reflect.ValueOf(m).Kind() == reflect.Ptr && reflect.ValueOf(m).IsNil()) {
```

Comment:
One of the helper sets only tested for `nil`, the other one also tested via reflection. To be safe, I keep the more thorough check.